MediaBrowser.Common/Json/Converters/JsonNullableStructConverter.cs

```diff
@@ -26,13 +27,27 @@ public JsonNullableStructConverter(JsonConverter<T?> baseJsonConverter)
         /// <inheritdoc />
         public override T? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
         {
-            // Handle empty string.
-            if (reader.TokenType == JsonTokenType.String && ((reader.HasValueSequence && reader.ValueSequence.IsEmpty) || reader.ValueSpan.IsEmpty))
+            if (reader.TokenType == JsonTokenType.String)
             {
-                return null;
-            }
+                // Handle empty string.
+                if ((reader.HasValueSequence && reader.ValueSequence.IsEmpty) || reader.ValueSpan.IsEmpty)
+                {
+                    return null;
+                }
+
+                string numberString = Uri.UnescapeDataString(reader.GetString());
```

Comment:
Why are you calling `UnescapeDataString` on the string `reader.GetString()` returns? What are you trying to achieve with that?

`GetString()` already returns an unescaped JSON string back to you.
https://docs.microsoft.com/en-us/dotnet/api/system.text.json.utf8jsonreader.getstring?view=netcore-3.1